sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -883,6 +890,184 @@ def _export_files(self, bq):
     return table.schema, metadata_list
 
 
+class _CustomBigQueryStorageStreamSourceBase(BoundedSource):
+  """A base class for BoundedSource implementations which read from BigQuery
+  using the BigQuery Storage API."""
+  def __init__(
+      self,
+      project=None,
```

Comment:
Dataset and Tables require default values to enable validation. The project value defaults to whatever project name is stored in the active configuration of the gcloud CLI (in case we don't specify it). I'm not sure if we should force the user to explicitly specify a project here. What do you think?

Reply:
Your explanation makes sense.

Reply:
if dataset and table are required it might make sense to move them up and not let them default to None to better document they are required.

Not sure of Beam's policies but you might want to use type annotations or documented expected types of each parameter.

Reply:
Added type annotations and documentation for the parameters.